MikuMikuLibrary.Native/TextureEncoderCore.cpp

```diff
@@ -1,243 +1,247 @@
+                Encode( image, encodedData.get(), UNCOMPRESSED_FORMAT );
+
+                if ( format == TextureFormat::RGB8 )
+                {
+                    for ( int i = 0; i < subTexture->Width; i++ )
+                    {
+                        for ( int j = 0; j < subTexture->Height; j++ )
+                        {
+                            byte* srcPixel = encodedData.get() + ( ( j * subTexture->Width ) + i ) * 4;
+                            byte* dstPixel = ( byte* ) dataPtr + ( ( j * subTexture->Width ) + i ) * 3;
+
+                            const Color8& color = *( Color8* ) srcPixel;
+
+                            dstPixel[ 0 ] = color.r;
+                            dstPixel[ 1 ] = color.g;
+                            dstPixel[ 2 ] = color.b;
+                        }
+                    }
+                }
+
+                else if ( format == TextureFormat::RGBA8 )
+                {
+                    for ( int i = 0; i < subTexture->Width; i++ )
+                    {
+                        for ( int j = 0; j < subTexture->Height; j++ )
+                        {
+                            byte* srcPixel = encodedData.get() + ( ( j * subTexture->Width ) + i ) * 4;
+                            byte* dstPixel = ( byte* ) dataPtr + ( ( j * subTexture->Width ) + i ) * 4;
+
+                            const Color8& color = *( Color8* ) srcPixel;
+
+                            dstPixel[ 0 ] = color.r;
+                            dstPixel[ 1 ] = color.g;
+                            dstPixel[ 2 ] = color.b;
+                            dstPixel[ 3 ] = color.a;
+                        }
+                    }
+                }
+            }
+
+            else
+            {
+                Encode( image, dataPtr, GetDXGIFormat( format ) );
+            }
+
+            return subTexture;
+        }
+
+        const TextureFormat format = GetTextureFormat( image.format );
+
+        if ( format == TextureFormat::Unknown )
+            throw gcnew Exception( "Unsupported compression type" );
+
+        SubTexture^ subTexture = gcnew SubTexture( ( int ) image.width, ( int ) image.height, format );
+
+        const pin_ptr<byte> dataPtr = &subTexture->Data[ 0 ];
+        memcpy( dataPtr, image.pixels, subTexture->Data->Length );
+
+        return subTexture;
+    }
+
+    Texture^ EncodeToTexture( const DirectX::ScratchImage& scratchImage, const TextureFormat formatHint )
+    {
+        const int cubeIndexMap[] = { 0, 1, 2, 3, 5, 4 };
+
+        const DirectX::TexMetadata& metadata = scratchImage.GetMetadata();
+
+        array<SubTexture^, 2>^ subTextures = gcnew array<SubTexture^, 2>( ( int ) metadata.arraySize, ( int ) metadata.mipLevels );
+        for ( size_t i = 0; i < metadata.arraySize; i++ )
+        {
+            size_t index;
+            if (metadata.arraySize == 6)
+                index = metadata.arraySize == 6 ? cubeIndexMap[i] : index;
+            else
+                index = i;

```

Comment:
The original bug is the uninitialized self assignment of `index` to itself (typo? - why does this even compile).

The added code now needlessly performs the check twice and adds an ugly non-const-able temp variable without removing the actual mistake.
The much simpler and preferred fix should be:
`const size_t index = metadata.arraySize == 6 ? cubeIndexMap[ i ] : i;`

The file also appears to be using a different encoding (?) hence messing up the file diff 
and does not follow the existing formatting convention of s p a c i n g everything a p a r t (   WeirdChamp on you @blueskythlikesclouds   )

Reply:
It's okay Sam, we all have incorrect opinions from time to time.